# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/DocumentLoadListener.cpp
# Commit: 0260b9ef8c6c
# Full Hash: 0260b9ef8c6cf2059f3024e798a2a58680ff9ee7
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2020-07-31 09:46:06
# Description:
#   Bug 1656039 - Fall back to content initiating the load if parent went away. r=mattwoodrow
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D85519
# ==============================================================================

diff -r 164f54ec7aff -r 0260b9ef8c6c netwerk/ipc/DocumentLoadListener.cpp
--- a/netwerk/ipc/DocumentLoadListener.cpp	Fri Jul 31 01:45:44 2020 +0000
+++ b/netwerk/ipc/DocumentLoadListener.cpp	Fri Jul 31 03:59:10 2020 +0000
@@ -894,7 +894,13 @@
   RefPtr<DocumentLoadListener> loadListener = do_QueryObject(parentChannel);
   registrar->DeregisterChannels(aLoadIdent);
 
-  MOZ_ASSERT(loadListener && loadListener->mOpenPromise);
+  if (!loadListener) {
+    // The parent went away unexpectedly.
+    *aListener = nullptr;
+    return nullptr;
+  }
+
+  MOZ_DIAGNOSTIC_ASSERT(loadListener->mOpenPromise);
   loadListener.forget(aListener);
 
   return (*aListener)->mOpenPromise;
